From 8a550063f6f13b13779e86c44e0a1dad7382f590 Mon Sep 17 00:00:00 2001
From: zhoupeng <zhoupeng.1996@bytedance.com>
Date: Fri, 30 Aug 2024 17:13:02 +0000
Subject: [PATCH] [AnchorPositioning] The anchor node is may be an inline node

This is causing a crash that became apparent in the in flight frontend
CL: crrev.com/c/5580046.

Crash id: 3060fd543745af42

Bug: 363034682
Change-Id: I0403e6392cd7d31d1e6f02d22f6d5278ea1bc211
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5826177
Reviewed-by: Alex Rudenko <alexrudenko@chromium.org>
Reviewed-by: Andrey Kosyakov <caseq@chromium.org>
Auto-Submit: Peng Zhou <zhoupeng.1996@bytedance.com>
Commit-Queue: Peng Zhou <zhoupeng.1996@bytedance.com>
Cr-Commit-Position: refs/heads/main@{#1349289}
---

--- a/third_party/blink/renderer/core/inspector/inspector_dom_agent.cc
+++ b/third_party/blink/renderer/core/inspector/inspector_dom_agent.cc
@@ -1755,7 +1755,7 @@
         "No layout object for node, perhaps orphan or hidden node");
   }
 
-  const auto* box = To<LayoutBox>(querying_object);
+  const auto* box = DynamicTo<LayoutBox>(querying_object);
   if (!box || !box->Container()) {
     return protocol::Response::ServerError(
         "The box or the container of the box does not exist");
